src/oca/go/src/goca/marketplaceapp.go

```diff
@@ -0,0 +1,172 @@
+package goca
+
+import "errors"
+
+// MarketPlaceApp represents an OpenNebula MarketPlaceApp
+type MarketPlaceApp struct {
+	XMLResource
+	ID   uint
+	Name string
+}
+
+// MarketPlaceAppPool represents an OpenNebula MarketPlaceAppPool
+type MarketPlaceAppPool struct {
+	XMLResource
+}
+
+// NewMarketPlaceAppPool returns a marketplace app pool. A connection to OpenNebula is
+// performed.
+func NewMarketPlaceAppPool(args ...int) (*MarketPlaceAppPool, error) {
+	var who, start, end int
+
+	switch len(args) {
+	case 0:
+		who = PoolWhoMine
+		start = -1
+		end = -1
+	case 1:
+		who = args[0]
+		start = -1
+		end = -1
+	case 3:
+		who = args[0]
+		start = args[1]
+		end = args[2]
+	default:
+		return nil, errors.New("Wrong number of arguments")
+	}
+
+    response, err := client.Call("one.marketapppool.info", who, start, end)
+	if err != nil {
+		return nil, err
+	}
+
+	marketapppool := &MarketPlaceAppPool{XMLResource{body: response.Body()}}
+
+	return marketapppool, err
+}
+
+// NewMarketPlaceApp finds a marketplace app object by ID. No connection to OpenNebula.
+func NewMarketPlaceApp(id uint) *MarketPlaceApp {
+	return &MarketPlaceApp{ID: id}
+}
+
+// NewMarketPlaceAppFromName finds a marketplace app object by name. It connects to
+// OpenNebula to retrieve the pool, but doesn't perform the Info() call to
+// retrieve the attributes of the marketplace app.
+func NewMarketPlaceAppFromName(name string) (*MarketPlaceApp, error) {
+	marketAppPool, err := NewMarketPlaceAppPool()
+	if err != nil {
+		return nil, err
+	}
+
+	id, err := marketAppPool.GetIDFromName(name, "/MARKETPLACEAPP_POOL/MARKETPLACEAPP")
+	if err != nil {
+		return nil, err
+	}
+
+	return NewMarketPlaceApp(id), nil
+}
+
+// CreateMarketPlaceApp allocates a new marketplace app. It returns the new marketplace app ID.
+// * tpl: template of the marketplace app
+func CreateMarketPlaceApp(tpl string) (uint, error) {
```

Comment:
It's needed the ID of the destination marketplace in which the app will be store.
```suggestion
func CreateMarketPlaceApp(tpl string, market int) (uint, error) {
```

Reply:
I agree. In that case we need to use the market in the client call too ;).
In the doc it is written: "IN | Int | The datastore ID." while it is the marketAPP id. That was wrong for me.

Reply:
I just fixed it

Reply:
Yes, you need to pass the parameter to the call below.
Thanks for notice about the docs, I'll fix that.

Reply:
Here you misplaced the type "int market" it would be "market int"

Reply:
Oh Indeed. I fixed that